docs/design/build-system.md

```diff
@@ -122,13 +155,13 @@ Large projects can also opt-out of AOT compilation for faster rebuilds.
 Angular CLI focuses on enabling tree-shaking (removing unused modules) and dead code elimination (removing unused module code).
 These two categories have high potential for size reduction because of network effects: removing code can lead to more code being removed.
 
-The main tool we use to achieve this goal are the compression capabilities of [Terser](https://github.com/terser/terser).
+The main tool we use to achieve this goal are the dead code elimination capabilities of [Terser](https://github.com/terser/terser).
 We also use Terser's mangling, by which names, but not properties, are renamed to shorter forms.
 The main characteristics of Terser to keep in mind is that it operates via static analysis and does not support the indirection introduced by module loading.
 Thus the rest of the pipeline is directed towards providing Terser with code that can be removed via static analysis in large single modules scopes.
 
 To this end we developed [@angular-devkit/build-optimizer](https://github.com/angular/angular-cli/tree/master/packages/angular_devkit/build_optimizer), a post-processing tool for TS code.
-Build Optimizer searches for code patterns produced by the TypeScript and Angular compiler that are known to inhibit dead code elimination, and converts them into equivalent structures that enable it instead.
+Build Optimizer searches for code patterns produced by the TypeScript and Angular compiler that are known to inhibit dead code elimination, and converts them into equivalent structures that enable it instead (the link above contains some examples).
```

Comment:
Addresses https://github.com/angular/angular-cli/pull/16400#discussion_r358404211. I didn't add an example because it would always be some 10 lines total. But the link above contains good examples right at start.